Formula/mailcatcher.rb

```diff
@@ -0,0 +1,176 @@
+class Mailcatcher < Formula
+  desc "Catches mail and serves it through a dream"
+  homepage "https://mailcatcher.me"
+  url "https://github.com/sj26/mailcatcher/archive/refs/tags/v0.8.1.tar.gz"
+  sha256 "c40105f63407efc3fb18c2b1e2408896e4b86b0c8ba4e53ea17e087cbb2d7fee"
+  license "MIT"
+
+  depends_on "pkg-config" => :build
+
+  uses_from_macos "curl" => :test
+  uses_from_macos "expect" => :test
+  uses_from_macos "netcat" => :test
+  uses_from_macos "libffi"
+  uses_from_macos "ruby"
+  uses_from_macos "sqlite"
+
+  on_linux do
+    depends_on "node" => :build
+  end
+
+  resource "eventmachine" do
+    url "https://rubygems.org/downloads/eventmachine-1.0.9.1.gem"
+    sha256 "9f4cb30b3bce0c2a90da875a81534f12cbf6f1174f80d64c32efbda1140b599e"
+  end
+
+  resource "mini_mime" do
+    url "https://rubygems.org/downloads/mini_mime-1.1.0.gem"
+    sha256 "30f2cca8a3c62b5c067f73a1834479dbd85d71f1291d65ffac933dc90796674d"
+  end
+
+  resource "mail" do
+    url "https://rubygems.org/downloads/mail-2.7.1.gem"
+    sha256 "ec2a3d489f7510b90d8eaa3f6abaad7038cf1d663cdf8ee66d0214a0bdf99c03"
+  end
+
+  resource "rack" do
+    url "https://rubygems.org/downloads/rack-1.6.13.gem"
+    sha256 "207e60f917a7b47cb858a6e813500bc6042a958c2ca9eeb64631b19cde702173"
+  end
+
+  resource "rack-protection" do
+    url "https://rubygems.org/downloads/rack-protection-1.5.5.gem"
+    sha256 "5a9f0d56ef96b616a242138986dc930aca76f6efa24f998e8683164538e5c057"
+  end
+
+  resource "tilt" do
+    url "https://rubygems.org/downloads/tilt-2.0.10.gem"
+    sha256 "9b664f0e9ae2b500cfa00f9c65c34abc6ff1799cf0034a8c0a0412d520fac866"
+  end
+
+  resource "sinatra" do
+    url "https://rubygems.org/downloads/sinatra-1.4.8.gem"
+    sha256 "18cb20ffabf31484b02d8606e450fbf040b52aea6147755a07718e9e0ffddd2f"
+  end
+
+  resource "daemons" do
+    url "https://rubygems.org/downloads/daemons-1.4.0.gem"
+    sha256 "c3fdae2a6309cef75fb98c9a39938fac058ac19cbe08dfd1a3d8297ca30a71ba"
+  end
+
+  resource "thin" do
+    url "https://rubygems.org/downloads/thin-1.5.1.gem"
+    sha256 "ea85c4c7d5b1bd29c4992757ccf8be0ddee9d4030f428db347f59a05474d3843"
+  end
+
+  resource "skinny" do
+    url "https://rubygems.org/downloads/skinny-0.2.4.gem"
+    sha256 "498d447cc99f638470b87fc5814bc82cd799cb9453f3a04523ff518f50df7ef8"
+  end
+
+  resource "sqlite" do
+    url "https://rubygems.org/downloads/sqlite3-1.4.2.gem"
+    sha256 "e8b8ef3b0f75c18e1a7ee62c5678c827e99389e53fa55eb7a9a5f57459004a52"
+  end
+
+  def install
+    if MacOS.version >= :mojave && MacOS::CLT.installed?
+      ENV["SDKROOT"] = ENV["HOMEBREW_SDKROOT"] = MacOS::CLT.sdk_path(MacOS.version)
+    end
+
+    # Someone else may have a better way of doing this, but this is what I could come up with
+    ENV["GEM_HOME"] = buildpath/"gem_home"
+    system "gem", "install", "--no-document", "bundler"
```

Comment:
I'm not a huge fan of doing `gem install` like this (though I think there might be other formulae that do something similar).

Would it make the formula too complicated if we just add a bundler `.gem` as a resource too?

Reply:
for that we will need a gemlock file, otherwise we have non-checksummed resources.

Reply:
Bear in mind that these gems are "throwaway."  We need bundler to do the `rake assets` and install all the required packages for dev but we don't keep these.  I can make it a resource, but then I need to not install it later in the block where we actually keep things around (where `ENV["GEM_HOME"] = libexec`.  I could also install a specific version of bundler if that's more kosher.

Reply:
As to where I got this from, it is what [licensefinder](Formula/licensefinder.rb) (in `test`) and [swiftgen](Formula/swiftgen.rb) (for install) do.  Swiftgen actually does almost the exact same thing I am, using `bundler` to generate some templates and then manually installing just the templates later.

Reply:
I'm aware that this is used only at build-time, and that other formulae do this too. A specific version of bundler would be good, though I think declaring it as a `resource` would be even better if it's not too complicated.

Reply:
Can do!  I'll do a revision today.

Reply:
resolved in f7a9e984bf2e18538507aecde5854d45ecb24658